### PR TITLE
net/stun: Fix STUN attribute padding

### PR DESCRIPTION
**Pull request by renthraysk**
Signed-off-by: RenThraysk <renthraysk@gmail.com>

---
This change is [Reviewable](https://reviewable.io/reviews/tailscale/tailscale/710)

### CODE REVIEW

File: net/stun/stun.go [hunk 1]
```diff
@@ -137,15 +137,15 @@ func foreachAttr(b []byte, fn func(attrType uint16, a []byte) error) error {
 		}
 		attrType := binary.BigEndian.Uint16(b[:2])
 		attrLen := int(binary.BigEndian.Uint16(b[2:4]))
-		attrLenPad := attrLen % 4
+		attrLenPad := (attrLen + 3) &^ 3
```
Comment (bradfitz) [inline net/stun/stun.go:140]: I'd prefer this be more explicit. The 3s and &^ bitmask is a bit arcane.

Let's just do:

```go
padLen := 4 - (attrLen % 4)
totalLen := attrLen + padLen
```

Then both the padding computation and the variable names are explicit.

Comment (bradfitz) [reply net/stun/stun.go:140]: A new test addition would also be good.

Comment (renthraysk) [reply net/stun/stun.go:140]: Your suggestion is incorrect.

It'll add 4 pad bytes when not needed.

**Comment by bradfitz** [comment]
Looks like the new tests fail, though?

**Comment by bradfitz** [comment]
Thanks!

**Pull request merged**